databrowser/src/pages/datasets/overview/OverviewListPage.vue

```diff
@@ -6,50 +6,171 @@ SPDX-License-Identifier: AGPL-3.0-or-later
+            {{
+              !isOtherDatasetsLoading && visibleDatasets?.length === 1
+                ? t('overview.listPage.datasetFound')
+                : t('overview.listPage.datasetsFound')
+            }}
+          </h1>
+          <InputCustom
+            v-model="filters.searchVal"
+            :placeholder="t('overview.listPage.searchDataset')"
+            type="search"
+            input-classes="w-full md:w-64"
+          />
+        </div>
+      </div>
 
-      <CardDivider />
+      <div class="flex w-full flex-col items-start gap-8 md:flex-row">
+        <!-- Mobile filters button -->
+        <div
+          class="flex w-full shrink-0 items-center gap-2 rounded border border-gray-300 px-3 py-2 font-semibold text-green-400 md:hidden"
+          @click="showFilters"
+        >
+          <div class="grow">
+            {{ t('overview.listPage.showFilters') }}
+          </div>
+          <div
+            v-if="appliedFiltersNum"
+            class="rounded bg-gray-200 px-2 text-sm text-gray-900"
+          >
+            {{ appliedFiltersNum }}
+          </div>
+          <IconFilter class="mr-2 h-3 w-3" />
+        </div>
+
+        <!-- Filters -->
+        <div
+          class="fixed inset-0 w-full shrink-0 rounded border border-gray-300 bg-white pb-3 md:relative md:block md:w-64"
+          :class="{
+            hidden: !isFiltersModalVisible,
+          }"
+        >
+          <!-- Title -->
+          <div
+            class="fixed top-0 z-10 flex w-full items-center gap-1 bg-white md:relative"
+          >
+            <h3
+              class="flex grow items-center gap-2 px-3 py-2 text-2xl font-semibold text-gray-900"
+            >
+              {{ t('overview.listPage.filter') }}
+              <div
+                v-if="appliedFiltersNum"
+                class="rounded bg-gray-200 px-2 text-sm text-gray-900"
+              >
+                {{ appliedFiltersNum }}
+              </div>
+            </h3>
+
+            <div
+              class="mx-3 hidden cursor-pointer select-none items-center justify-center gap-1 rounded border border-gray-300 px-2 py-1 text-sm text-green-400 md:flex"
+              @click="resetFilters"
+            >
+              <IconDelete class="h-4 w-4 text-error" />
+              {{ $t('overview.listPage.resetFilters') }}
+            </div>
+            <div
+              class="mr-3 flex h-5 w-5 items-center justify-center border border-gray-300 text-green-400 md:hidden"
+              @click="resetFilters"
+            >
+              <IconClose class="h-4 w-4" />
+            </div>
+          </div>
+
+          <!-- Filters list -->
+          <div class="h-full overflow-y-auto py-14 md:h-auto md:p-0">
+            <div
+              class="truncate border-t border-gray-300 px-3 py-2 text-dialog"
+              @click="toggleFilter('hasNoMetadata')"
+            >
+              <ToggleCustom v-model="_inputModels.no_metadata" class="mr-2" />
```

Comment:
From my perspective, using the array filters to also handle the v-model of the checkboxes would only further complicate the code and make it even less performant because it would dynamically need to search in the filters array to determine whether the checkbox value is true or not. I've optimized the management of the _inputModels array by creating a new method called setInputModelKeyValue, making everything more robust and maintainable. Regarding the comment below, for this optimization, I would also avoid refactoring the code to update _inputModels all at once, as it would make it less readable, and there is currently no performance concern.

Reply:
@MatteoBiasi my main concern is about having contradicting states. First, a developer must take care to not mismatch the variables for `toggleFilter` and `v-model`. Second, by design there is always a short period of time, where the states mismatch, because the `click` handler and `v-model` changes cannot execute at the same time. This can lead to really nasty bugs that are hard to track (race-conditions).

My gut feeling tells me that the components state management should be revised (see also the other issue about `_inputModels` update). On the other hand, the feature works as expected and if one day we'll find out that we have to improve, then we should be able to do it without greater struggle. So let's keep the code as provided now.